Link SQL exports to Django view

NoahCarnahan opened the pull request:
@benrudolph @snopoke cc @kaapstorm 
This PR adds a django view that can be used to download exports in the new format. This is a very minimal implementation, that is still missing a lot of functionality, like filters and case exports. The goal here was mainly to figure out which pieces are missing before everything can be wired together.

NoahCarnahan committed:
Add format to ExportFile

NoahCarnahan committed:
Fix bad TableConfiguration.name refs

NoahCarnahan committed:
Add function for starting export task

_get_export_download() is based off of cache_file_to_be_served() and export_async()

NoahCarnahan committed:
Add view for initiating download with new export backend

snopoke committed:
Merge branch 'master' into wire-sql-export-to-ui

snopoke commented:
Merged master to force travis build

dimagimon commented on corehq/apps/export/tasks.py:
```diff
@@ -0,0 +1,26 @@
+
+
+@task
+def get_export_download(export_instances, filters, download_id, filename=None, expiry=10*60*60):
```
E226 missing whitespace around arithmetic operator

dimagimon commented on corehq/apps/export/views.py:
```diff
@@ -1417,3 +1429,34 @@ def commit(self, request):
+
+        return get_download_task(
+            export_instances=[export_instance],
+            filters=[], # TODO: Do something with export_filters
```
E261 at least two spaces before inline comment

NoahCarnahan committed:
[ci skip] lint

NoahCarnahan committed:
Rename functions

NoahCarnahan commented:
@snopoke updated

NoahCarnahan committed:
Improve code reuse in DownloadNewFormExportView

benrudolph commented:
:+1: :mans_shoe: :construction:

NoahCarnahan committed:
Add more export filters

NoahCarnahan committed:
Create separate Couch/ES versions of export filter forms

NoahCarnahan committed:
Add partialy written tests

dimagimon commented on corehq/apps/export/tests/test_export_forms.py:
```diff
@@ -0,0 +1,32 @@
+        form = FilterFormESExportDownloadForm(Domain(name="foo"), pytz.utc, form_data)
+        self.assertTrue(form.is_valid())
+        raise NotImplementedError
+
```
W391 blank line at end of file

NoahCarnahan commented:
Made some adjustments to the views to use the new filter classes. Didn't have time to finish the tests this afternoon, but will work on them monday.

NoahCarnahan committed:
Implement simple tests for ES form export filter form

dimagimon commented on corehq/apps/export/tests/test_export_forms.py:
```diff
@@ -0,0 +1,62 @@
+            'group': '',
+            'date_range': '2015-06-25 to 2016-02-19',
+        }
+        def mock_users_matching_filter(domain, user_filters):
```
E301 expected 1 blank line, found 0

dimagimon commented on corehq/apps/export/tests/test_export_forms.py:
```diff
@@ -0,0 +1,62 @@
+        with patch("corehq.apps.export.forms.users_matching_filter", mock_users_matching_filter):
+            user_filter = form._get_user_filter()
+        self.assertEqual(user_filter.submitted_by, [None, "some_user_id", "some_other_user_id"])
+
```
W391 blank line at end of file

NoahCarnahan committed:
[ci skip] lint

NoahCarnahan committed:
[ci skip] Remove comment

NoahCarnahan committed:
[ci skip] remove comment

NoahCarnahan committed:
Set format on export writers

NoahCarnahan committed:
Avoid DB access in test

Domain.uses_locations accesses the database,
using a dummy object with an interface like Domain fixes that.

NoahCarnahan commented:
:white_check_mark:

dimagimon commented on corehq/apps/export/tests/test_export_forms.py:
```diff
@@ -0,0 +1,68 @@
+from django.test import SimpleTestCase
+from mock import patch
+
+from corehq.apps.domain.models import Domain
```
F401 'Domain' imported but unused

dimagimon commented on corehq/apps/export/tests/test_export_forms.py:
```diff
@@ -0,0 +1,68 @@
+
+    def setUp(self):
+        DomainObject = namedtuple('DomainObject', ['uses_locations', 'name'])
+        self.project =  DomainObject(False, "foo")
```
E222 multiple spaces after operator

benrudolph commented:
sweet, maybe just cleanup the last dimagimon and then good from my end

NoahCarnahan committed:
[ci skip] lint

NoahCarnahan commented:
(fixed)
